### PR TITLE
Migrate TestRatingManager.js

### LINKED ISSUE #11232: Migrate /components/TestRatingManager.js to react-testing-library
Migrate /components/TestRatingManager.js to react-testing-library

### PR DESCRIPTION
**Pull request by bobsilverberg**
~~Depends on #11429~~
Fixes #11232

### CODE REVIEW

File: tests/unit/amo/components/TestRatingManager.js [hunk 1]
```diff
@@ -84,471 +103,321 @@ describe(__filename, () => {
...
   it('passes review=null when no saved review exists', () => {
-    const { addon, store } = createStoreWithLatestReview({ review: null });
-    const root = render({ addon, store });
+    renderWithReview({ review: null });
 
     // This exits the loading state.
-    expect(root.find(UserRating)).toHaveProp('review', null);
-  });
-
-  it('configures a rating component', () => {
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestRatingManager.js:200]: Note that this test has been replaced with other tests that verify and interact with Ratings.

File: tests/unit/amo/components/TestRatingManager.js [hunk 2]
```diff
@@ -84,471 +103,321 @@ describe(__filename, () => {
...
-      'onSelectRating',
-      root.instance().onSelectRating,
-    );
-    expect(root.find(UserRating)).toHaveProp(
-      'review',
-      selectReview(store.getState().reviews, review.id),
-    );
-  });
-
-  it('passes an add-on to the report abuse button', () => {
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestRatingManager.js:216]: Note that the entire suite for `ReportAbuseButton` will be moved into this file (already done, but saving for a separate patch to make things more manageable for review) so this test will not be needed.

File: tests/unit/amo/components/TestRatingManager.js [hunk 3]
```diff
@@ -84,471 +103,321 @@ describe(__filename, () => {
...
+      renderWithReview({ addon, signIn: false });
+
+      expect(
+        screen.getByRole('link', { name: `Log in to rate this ${prompt}` }),
+      ).toBeInTheDocument();
     });
   });
 
   describe('inline features', () => {
-    it('configures AddonReviewManagerRating when beginningToDeleteReview', () => {
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestRatingManager.js:365]: These two tests are dealt with in more integration style tests above.

File: tests/unit/amo/components/TestRatingManager.js [hunk 4]
```diff
@@ -84,471 +103,321 @@ describe(__filename, () => {
...
-      const { addon, store } = createStoreWithLatestReview({ review });
-      const root = render({ addon, store });
-
-      const reviewCard = root.find(AddonReviewCard);
-      expect(reviewCard).toHaveProp('review', createInternalReview(review));
-    });
+      const body = 'This is some review copy';
+      renderWithReview({ review: { ...fakeReview, body } });
 
-    it('shows controls by default', () => {
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestRatingManager.js:455]: These tests for showing and hiding controls are covered by some tests above that exercise the component.

**Review by diox — APPROVED** [review]

**Pull request merged**